sql/parser/builtins.go

```diff
@@ -951,6 +969,42 @@ func round(x float64, n int64) (Datum, error) {
 	return DFloat(y), err
 }
 
+// Pick the greatest (or least value) from a tuple.
+func pickFromTuple(greatest bool, args DTuple) (Datum, error) {
+	pos := -1
+	// Pick the first non-NULL value.
+	for i, d := range args {
+		if d != DNull {
+			pos = i
+			break
+		}
+	}
+	if pos == -1 {
+		return DNull, nil
+	}
+	g := args[pos]
+	// Pick a greater (or smaller) value.
+	for _, d := range args[pos+1:] {
+		if d == DNull {
```

Comment:
I think you can get rid of this check since `evalComparison` will return `DNull` if either argument is `DNull`. Similarly, you might be able to remove the initial loop above looking for the first non-NULL element, though I'm less certain whether the other changes necessary to make that work would be worth it.